src/System.Text.Formatting/System/Text/Formatting/IFormatter.cs

```diff
@@ -8,14 +8,20 @@ namespace System.Text.Formatting
     // this interface would be implemented by types that want to support formatting, i.e. TextWriter/StringBuilder-like types.
     // the interface is used by an extension method in IFormatterExtensions.
     // One thing I am not sure here is if it's ok for these APIs to be synchronous, but I guess I will wait till I find a concrete issue with this.
-    public interface IFormatter
+    public interface IStream
     {
-        Span<byte> FreeBuffer { get; }
-        void CommitBytes(int bytes);
+        Span<byte> AvaliableBytes { get; }
```

Comment:
@davidfowl, I cannot answer your question before I know what we are planing to do in IFormatterExtensions.Append. It seems to me like the Append methods cannot be async, and so I am not sure FlushAsync makes sense on this interface. 

I prototypes some things yesterday, and currently I am trying to change the Append methods to TryAppend and then build spinning sync Appends and async AppendAsync on top of these TryAppends. I still have some problems with it, but it looks promising.